biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -1751,6 +1755,23 @@ void augmentImports(Packages imports, Packages exports) throws Exception {
 		}
 	}
 
+	void moveDynamicImports() {
+		Parameters dynamicImportPackages = getDynamicImportPackage();
+		Iterator<Entry<PackageRef,Attrs>> importsIterator = imports.entrySet().iterator();
+		while (importsIterator.hasNext()) {
+			Entry<PackageRef,Attrs> packageEntry = importsIterator.next();
+			PackageRef packageRef = packageEntry.getKey();
+			Attrs attrs = packageEntry.getValue();
+			String resolution = attrs.get(Constants.RESOLUTION_DIRECTIVE);
+			if (Constants.DYNAMIC_ATTRIBUTE.equals(resolution)) {
+				attrs.remove(Constants.RESOLUTION_DIRECTIVE);
+				dynamicImportPackages.add(packageRef.getFQN(), attrs);
+				importsIterator.remove();
```

Comment:
I'll update the PR, but as far I could see plugins can not run at this point of the execution. They are invoked earlier.